### PR TITLE
Basic backend and frontend service infrastructure

### PR DESCRIPTION
**Pull request by mhauru**
Adds
* Relevant database tables
* Backend API endpoints
* Frontend pages for adding, removing, running, and editing services.

No scheduling yet.

Builds on #207, so getting that merged first will make the diff manageable.

### CODE REVIEW

**Comment by mhauru** [comment]
I tagged y'all as reviewers, but maybe hold off until #207 is merged, because right now the diff is entirely unworkable.

**Comment by EdwinB12** [comment]
image

@mhauru - Could you quickly talk through what you expect from the user? Service Name is explanatory, but URL? Does this mean the service needs to call an API?

**Comment by mhauru** [comment]
Yes, the URL is the URL to which a REST request should be sent when the service is to be run. E.g. the URL for triggering an Azure function to run.

File: dtbase/tests/test_services.py
Comment (EdwinB12) [inline dtbase/tests/test_services.py:301]:
```suggestion
    Test the run_service function. 
    
    The steps in this test are:
    
     1. Insert parameter sets
     2. Mock Service 1 and Service 2
     3. Run Service 1 four times, and Service 2 once with different parameters 
     4. Assert lenth of returned run list for each Service
     5. Assert length of run list for Service 1 with a certain parameter combination
     6. Assert length of run list for all runs 
     7. Assert the contents of each run in run list
```

**Review by EdwinB12 — COMMENTED** [review]
Couple of small comments, suggestions and clarifications. 

I've tested it thoroughly locally and it seems to work as expected as far as I can tell. Good work!

File: dtbase/tests/test_services.py
Comment (mhauru) [reply dtbase/tests/test_services.py:1]: Is there a particular thing you want to avoid confusing `test_services` with?

If we rename this, I would also rename the corresponding test modules `test_users`, `test_locations`, etc.

Hadn't thought about it, but maybe splitting the tests into frontend, database, backend api, and others would be good.

File: dtbase/tests/test_services.py
Comment (mhauru) [reply dtbase/tests/test_services.py:74]: It's just checking that the `insert_service` function runs without raising an error. I can't really think of what to assert here.

File: dtbase/tests/test_services.py
Comment (EdwinB12) [reply dtbase/tests/test_services.py:1]: Yer, I usually reflect the packages structure as closely as I can in the tests. But I don't think its too important at this point, its clear how it is.

**Review by EdwinB12 — APPROVED** [review]

**Pull request merged**
